### PR TITLE
Add bleeding butchery action

### LINKED ISSUE #45739: Blood harvesting (in relation to Magiclysm mod)
# How often do you see this screen pop up?
5bc8dae1c49c2ddf1597a0ca1ad618eb

For many of us - way too much. While blood procurement is fine the UI handling of it deserves better.

# Describe the solution you'd like
@Fris0uman suggested, and I agree completely, to move corpse blood harvesting into a specialised option akin to skinning.

# Additional context
Butchering without draining blood would yield usual drops, sans blood, which would remove unnecessary clutter. Or for additional immersion, the blood can be instantly spilled on the ground and unrecoverable 

The aforementioned solution can be incorporated into the main game and provide a lot of features down the line.
- NPCs (dead or alive) can be drained of healthy blood, which can be used for blood infusions. This can allow bleeding wounds instead of damaging limbs to use something similar to health mechanic in terms of blood.
- Blood (Human/Animal/Zombie) can be used as an ingredient for cooking or chemistry.
- Expanded transfusion\injection mechanics (develop serums\antibodies, perform experiments banned by geneva convention etc)

# Describe alternatives you've considered
- Remove blood altogether and use some solid drops as magiclysm ingredients.
- Drop tainted blood in blood sacks 
- Make blood valuable enough that players would anticipate the additional menu pop up

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary

SUMMARY: Features "Add bleeding butchery action"

#### Purpose of change
Fixes #45739 

#### Describe the solution

Adds an action to bleed corpse to be able to get blood if the player is interested by it and have regular butchery just ignore blood

#### Testing
- Load vanilla world without error
- Kil creatures of various sizes
- Can bleed small creature with just a knife
- Big creatuer require rope and tree or butchering rack
- Actually get blood from bleeding
- Can only bleed creature once
- Can't bleed quartedred corpses or dress field corpses
- Don't get prompt for handling blood if oding somethign else than bleeding > blood is dropped on floor automatically

- Load magiclysm without error
- Can do all the above without error
- Can still use Blood power generator cbm with tainted and dragon blood

#### Additional context
Blood Generator Cbm can't use regular, human and demi human blood though because they are now food and untill the fuel consumption menu is fixed (https://github.com/CleverRaven/Cataclysm-DDA/issues/45620) food can't be a fuel for CBM.

I didn't add any recip to use blood because I don't have much time to work on this so they'll have to come in another PR from someone else.

### CODE REVIEW

**Comment by Venera3** [comment]
Survivor blutwurst, here we come!

**Comment by Mrzjadacz** [comment]
What about acidic and gasoline zombies or pupating crawlers?

**Comment by Venera3** [comment]
For the more exotic critters/substances it might make sense to use more neutral language ( "extract liquids", for example). That opens the door to extracting gall, haemolymph or other potential chems, and would allow me to do away with those weird venom glands and let you get the uncut liquid directly.

**Comment by Shodan14** [comment]
Really cool change, kind of feels like you'd need some special tool(s) for this beyond the normal butchery stuff though along the same lines as scalpel for dissection.

**Comment by Fris0uman** [comment]
Pretty sure a sharp nife is the only thing you need, and something to suspend the body if it's big

**Comment by Venera3** [comment]
Maybe a funnel to collect the liquids?

**Comment by Fris0uman** [comment]
Yeah maybe a funnel or a bucket could make sense

File: data/json/items/comestibles/carnivore.json [hunk 1]
```diff
@@ -123,6 +123,22 @@
     "flags": [ "SMOKABLE", "RAW", "PREDATOR_FUN" ],
     "smoking_result": "meat_smoked"
   },
+  {
+    "id": "animal_blood",
+    "type": "COMESTIBLE",
+    "copy-from": "meat",
+    "name": { "str": "blood", "str_pl": "blood" },
```
Comment (actual-nh) [inline data/json/items/comestibles/carnivore.json:130]:
```suggestion
    "name": { "str_sp": "blood" },
```

File: data/json/items/comestibles/carnivore.json [hunk 2]
```diff
@@ -246,6 +278,22 @@
     "description": "Freshly butchered from the body of a heavily mutated creature that was unsettlingly humanoid in appearance.  It has odd bits of fur and other tissue lodged in it that clearly don't belong.  You'd have to be crazy or starving to eat this.",
     "material": [ "hflesh" ]
   },
+  {
+    "id": "mutant_human_blood",
+    "type": "COMESTIBLE",
+    "copy-from": "mutant_human_flesh",
+    "name": { "str": "mutant human blood", "str_pl": "mutant human blood" },
```
Comment (actual-nh) [inline data/json/items/comestibles/carnivore.json:285]:
```suggestion
    "name": { "str_sp": "mutant human blood" },
```

File: data/json/items/comestibles/carnivore.json [hunk 3]
```diff
@@ -827,6 +874,20 @@
     "flags": [ "TRADER_AVOID", "SMOKABLE", "NUTRIENT_OVERRIDE", "RAW" ],
     "smoking_result": "dry_meat_tainted"
   },
+  {
+    "id": "blood_tainted",
+    "type": "COMESTIBLE",
+    "copy-from": "meat_tainted",
+    "name": { "str": "tainted blood", "str_pl": "tainted blood" },
```
Comment (actual-nh) [inline data/json/items/comestibles/carnivore.json:881]:
```suggestion
    "name": { "str_sp": "tainted blood" },
```

File: data/json/items/comestibles/carnivore.json [hunk 4]
```diff
@@ -1204,6 +1265,22 @@
     "material": [ "flesh" ],
     "flags": [ "STRICT_HUMANITARIANISM" ]
   },
+  {
+    "id": "demihuman_blood",
+    "type": "COMESTIBLE",
+    "copy-from": "demihuman_flesh",
+    "name": { "str": "demihuman blood", "str_pl": "demihuman blood" },
```
Comment (actual-nh) [inline data/json/items/comestibles/carnivore.json:1272]:
```suggestion
    "name": { "str_sp": "demihuman blood" },
```

File: src/activity_handlers.cpp [hunk 5]
```diff
@@ -911,7 +924,7 @@ static void butchery_drops_harvest( item *corpse_item, const mtype &mt, player &
                 continue;
             }
         }
-        // field dressing ignores everything outside below list
+        // field dressing ignores skin flesh, and blood
```
Comment (actual-nh) [inline src/activity_handlers.cpp:927]:
```suggestion
        // field dressing ignores skin, flesh, and blood
```

File: src/activity_handlers.cpp [hunk 6]
```diff
@@ -1234,7 +1258,8 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
     // and the liquid handling was interrupted, then the activity was canceled,
     // therefore operations on this activities targets and values may be invalidated.
```
Comment (actual-nh) [inline src/activity_handlers.cpp:1259]:
```suggestion
    // therefore operations on this activity's targets and values may be invalidated.
```

**Review by actual-nh — COMMENTED** [review]
Interesting!

**Comment by actual-nh** [comment]
I have some suspicion that already-bled meat would take less time to salt down or smoke (part of the time for these processes may include bleeding it out anyway, at least if fresh enough).

**Comment by EvgenijM86** [comment]
If you bleed a corpse and then butcher it then the corpse will still drop the same amount of blood. I tested it on beavers with latest commit at the time of writing this.

**Comment by Fris0uman** [comment]
Please open an issue about it so the bug doesn't get lost

**Pull request merged**